gcovr/writer/json.py

```diff
@@ -58,6 +57,26 @@
 PRETTY_JSON_INDENT = 4
 
 
+get_filename = None
+
+
+def set_filename_function(options):
+    if options.json_base:
+        options.json_base = os.path.normpath(options.json_base).replace("\\", "/")
```

Comment:
Should this replace `\\`, or only `os.path`?

Reply:
The backslash shall be replaced with forward slash. This is done in several places, maybe better create a function `gcovr_normalize`.

Reply:
A quick grep for this code pattern yields the following occurrences:

```none
gcovr/configuration.py:            value = str(self.value).replace("\\", os.sep).replace("/", os.sep)
gcovr/gcov.py:    gcovname = gcovname.replace("/", os.sep)
gcovr/tests/test_gcovr.py:    contents = contents.replace("\\", "/")
gcovr/tests/test_gcovr.py:    contents = contents.replace("\\", "/")
gcovr/utils.py:    return path.replace(os.path.sep, "/")
gcovr/utils.py:    return normalized.replace("\\", "/")
gcovr/writer/cobertura.py:        package.set("name", packageName.replace("/", "."))
gcovr/writer/cobertura.py:    etree.SubElement(sources, "source").text = options.root.strip().replace("\\", "/")
gcovr/writer/html.py:        return "." if self.directory == "" else self.directory.replace("\\", "/")
gcovr/writer/html.py:        ).replace("\\", "/")
gcovr/writer/html.py:    filename = filename.replace(os.sep, "/")
gcovr/writer/txt.py:        fh.write("Directory: " + options.root.replace("\\", "/") + "\n")
noxfile.py:    session.env["CC"] = shutil.which(session.env["CC"]).replace(os.path.sep, "/")
noxfile.py:    session.env["CXX"] = shutil.which(session.env["CXX"]).replace(os.path.sep, "/")
noxfile.py:    ).replace(os.path.sep, "/")
```

Replacing `\` to `/` is fine when done for display purposes, i.e. if it is not necessary to resolve the path afterwards. But to be ultra-pedantic, `\` is a legal character for filenames on Linux, so replacing it could technically break some use cases.

But it seems that `presentable_filename()` does exactly this transformation as well, and is used in the JSON format, so that your suggested code is in line with how things currently work.

Reply:
I can change this with a separate PR to only replace this on Windows. Than `os.path.sep` is correct.